swift/Glacier2/callback/Sources/Client/main.swift

```diff
@@ -0,0 +1,188 @@
+//
+// Copyright (c) ZeroC, Inc. All rights reserved.
+//
+
+import Foundation
+import Glacier2
+import Ice
+import PromiseKit
+
+class CallbackReceiverI: CallbackReceiver {
+    func callback(current _: Current) throws {
+        print("Received callback")
+    }
+}
+
+enum Option: String {
+    case twoway = "t"
+    case oneway = "o"
+    case batchOneway = "O"
+    case flush = "f"
+    case overrideContextField = "v"
+    case fakeCategory = "F"
+    case shutdown = "s"
+    case exit = "x"
+    case help = "?"
+}
+
+func menu() {
+    print(
+        """
+        usage:
+            t: invoke callback as twoway
+            o: invoke callback as oneway
+            O: invoke callback as batch oneway
+            f: flush all batch requests
+            v: set/reset override context field
+            F: set/reset fake category
+            s: shutdown server
+            x: exit
+            ?: help
+        """)
+}
+
+func run() -> Int32 {
+    do {
+        var args = CommandLine.arguments
+
+        let communicator = try Ice.initialize(args: &args, configFile: "config.client")
+        defer {
+            communicator.destroy()
+        }
+
+        guard args.count == 1 else {
+            print("too many arguments")
+            return 1
+        }
+
+        guard let defaultRouter = communicator.getDefaultRouter(),
+            let router = try checkedCast(prx: defaultRouter, type: Glacier2.RouterPrx.self) else {
+            return 1
+        }
+
+        // Prompt user for authentication
+        var session: SessionPrx?
+        while session == nil {
+            print("This demo accepts any user-id / password combination.")
+            print("Username: ", terminator: "")
+            let username = readLine(strippingNewline: true)!
+            print("Password: ", terminator: "")
+            let password = readLine(strippingNewline: true)!
+
+            // Create session
+            do {
+                session = try router.createSession(userId: username, password: password)
+                break
+            } catch {
+                print("Cannot created session: \(error)")
+            }
+        }
+
+        guard let acmTimeout = try? router.getACMTimeout(),
+            let connection = router.ice_getCachedConnection() else {
+            return 1
+        }
```

Comment:
I don't thin we should be using `try?` here or `guard` for `acmTimeout`
```
let acmTimeout = try router.getACMTimeout()
guard let connection = router.ice_getCachedConnection() else {
    fatalError()
    return 1
}
```

Reply:
Yes this works better. However don't need the `return 1` as it will never be executed if `fatalError()` is called.